### PR TITLE
Intake

### PR DESCRIPTION
**Pull request by Berdenson**
You can press the A button to run the intake. (intake motor speed may need to be flipped, but that's for later)

### CODE REVIEW

**Review by QuackitsQuinn — DISMISSED** [review]
looks good, thank you for documenting everything.

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -36,4 +36,12 @@ public static class Drive {
     /** The max speed the robot can rotate */
     public static double MAX_ANGULAR_SPEED = Math.PI / 2;
   }
+
+  public static class Intake {
+    /** Motor id of the Intake motor. */
+    public static final int INTAKE_MOTOR_ID = 0;
+
```
Comment (willitcode) [inline src/main/java/frc/robot/Constants.java:43]: Will want a DIO port ID for the limit switch here somewhere.

File: src/main/java/frc/robot/subsystems/Intake.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj.motorcontrol.Spark;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+/** The intake of the robot. */
+public class Intake extends SubsystemBase {
+
+  private Spark intake;
```
Comment (willitcode) [inline src/main/java/frc/robot/subsystems/Intake.java:21]: Needs to be a CANSparkMax in Brushless mode

File: src/main/java/frc/robot/subsystems/Intake.java
Comment (willitcode) [inline src/main/java/frc/robot/subsystems/Intake.java:1]: Add a method somewhere in this class to get the state of the limit switch

File: src/main/java/frc/robot/commands/RunIntake.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    intake.stop();
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    return false;
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/RunIntake.java:46]: We have two options here. Either return the state of the limit switch from this method, or (better yet) we create this command and a command that ends when the limit switch is `true`, and put them in a ParallelRaceGroup. You need to try the ParallelRaceGroup method first and see if it causes an error where the commands conflict with each other—simulation mode is your bestie (or maybe you can talk electrical into plugging a Spark Max and a limit switch into the test board for you, idk).

**Review by willitcode — CHANGES_REQUESTED** [review]
See individual code comments.

File: src/main/java/frc/robot/commands/RunIntake.java (hunk 3 above)
Comment (willitcode) [reply src/main/java/frc/robot/commands/RunIntake.java:46]: Okay, decision has been made: do the first option. Just return the state of the limit switch from the `isFinished()` method of `RunIntake`.

**Review by willitcode — CHANGES_REQUESTED** [review]
Looks good! Thanks!

File: src/main/java/frc/robot/commands/RunIntake.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.Intake;
+
+public class RunIntake extends Command {
+
+  private Intake intake;
+
+  /** Creates a new RunIntake. */
+  public RunIntake() {
+    this.intake = RobotContainer.intake;
```
Comment (willitcode) [inline src/main/java/frc/robot/commands/RunIntake.java:24]: call `addRequirements(Command...)`

**Review by willitcode — CHANGES_REQUESTED** [review]
Sorry...

**Review by willitcode — APPROVED** [review]
Looks good! Thanks! Sorry about all the change-request chaos.

**Pull request merged**